server/szurubooru/func/posts.py

```diff
@@ -967,3 +972,81 @@ def search_by_image(image_content: bytes) -> List[Tuple[float, model.Post]]:
         ]
     else:
         return []
+
+
+PoolPostsAround = namedtuple('PoolPostsAround', 'pool first_post prev_post next_post last_post')
+
+def get_pool_posts_around(post: model.Post) -> List[PoolPostsAround]:
+    around = dict()
+    pool_ids = set()
+    post_ids = set()
+
+    dbquery = """
+    SELECT around.ord, around.pool_id, around.post_id, around.delta
+    FROM pool_post pp,
+        LATERAL get_pool_posts_around(pp.pool_id, pp.post_id) around
+    WHERE pp.post_id = :post_id;
+    """
```

Comment:
Use SQLAlchemy instead of writing raw SQL.

Reply:
The `search_by_image()` function uses raw SQL only because Postgres' `unnest()` function is not implemented in SQLAlchemy. But this query should be able to be written using SQLAlchemy

Reply:
You may want to also look at how we currently implement `GET /post/<ID>/around` as a springboard